### PR TITLE
Add 3-qubit Cirq gates.

### PR DESCRIPTION
**Pull request by sergeisakov**
This PR also adds multi-qubit identity and matrix gates.

### CODE REVIEW

File: tests/gates_cirq_testfixture.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ namespace qsim {
 namespace CirqCircuit1 {
 
 template <typename fp_type>
-Circuit<Cirq::GateCirq<fp_type>> GetCircuit(bool qsimh) {
+Circuit<Cirq::GateCirq<fp_type>> GetCircuit(bool qsim) {
```
Comment (95-martin-orion) [inline tests/gates_cirq_testfixture.h:30]: Was the parameter name here previously incorrect?

File: qsimcirq/qsim_circuit.py [hunk 2]
```diff
@@ -92,14 +94,26 @@ def _cirq_gate_kind(gate: cirq.ops.Gate):
...
+    if gate.exponent == 1 and gate.global_shift == 0:
+      return qsim.kCCZ
+    return qsim.kCCZPowGate
+  if isinstance(gate, cirq.ops.CCXPowGate):
+    if gate.exponent == 1 and gate.global_shift == 0:
+      return qsim.kCCX
+    return qsim.kCCXPowGate
+  if isinstance(gate, cirq.ops.CSwapGate):
+    return qsim.kCSwapGate
   if isinstance(gate, cirq.ops.MatrixGate):
```
Comment (95-martin-orion) [inline qsimcirq/qsim_circuit.py:111]: [No action required] With some extra tinkering in the python code, I think we can convert any non-decomposable Cirq gates into their matrix forms, allowing qsim to execute _any_ Cirq circuit with gates of up to 6 qubits.

I can take a look at this if you aren't already working on it.

**Review by 95-martin-orion — APPROVED** [review]
Very cool! Just a couple point to clarify before merging.

File: tests/gates_cirq_testfixture.h (hunk 1 above)
Comment (sergeisakov) [reply tests/gates_cirq_testfixture.h:30]: Thank you for the review. Yes, the parameter name here was incorrect.

File: qsimcirq/qsim_circuit.py (hunk 2 above)
Comment (sergeisakov) [reply qsimcirq/qsim_circuit.py:111]: I'm not working on it. It would be great if you take a look at this.

**Pull request merged**